FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +30,34 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: self.url) { [weak self] result in
+			guard let _ = self else { return }
+
+			guard case .success = result else {
+				completion(.failure(Error.connectivity))
+				return
+			}
+
+			guard let (data, httpResponse) = try? result.get(), httpResponse.statusCode == 200 else {
+				completion(.failure(Error.invalidData))
+				return
+			}
```

Comment:
Instead of checking twice if `result` represents a `success` (with `case .success = result` and later with `try? result.get()`, it's possible to perform this operation once:
```suggestion
			guard let (data, httpResponse) = try? result.get() else {
				return completion(.failure(Error.connectivity))
			}

			guard httpResponse.statusCode == 200 else {
				return completion(.failure(Error.invalidData))
			}
```
Another possibility would be to use a `switch` statement on the result, and extract the map logic to a helper function, as shown in the program:
```swift
	switch result {
	case let .success((data, response)):
		completion(RemoteFeedLoader.map(data, from: response))
	case .failure:
		completion(.failure(Error.connectivity))
	}
```
That way, we can separate `loading` logic (calling `client.get` and dealing with its `result`) from `mapping` logic (converting this result into a `FeedLoader.Result`)